gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -178,6 +210,81 @@ public boolean accept(File file) {
   }
 
   /**
+   * Recursively load job configuration files under given URI of **directory of config files folder**
+   */
+  private static void loadGenericJobConfigsRecursive(List<Properties> jobConfigs, Properties rootProps,
+      Set<String> jobConfigFileExtensions, Path configDirPath)
+      throws ConfigurationException, IOException {
+
+    /* create the generic file system */
+    Configuration conf = new Configuration();
+    try (FileSystem filesystem = configDirPath.getFileSystem(conf)) {
+      if (!filesystem.exists(configDirPath)) {
+        throw new RuntimeException("No specified .properties directory found :" + configDirPath.toString());
+      }
+
+      FileStatus[] propertiesFilesStatus = filesystem.listStatus(configDirPath, PROPERTIES_PATH_FILTER);
```

Comment:
When you run `fileSystem.listStatus`, always check that the result is not null. The contract of this method allows it to return null in certain situations, usually throwing `NullPointerExceptions` afterwards.